orderer/consensus/etcdraft/util.go

```diff
@@ -253,16 +255,176 @@ func CheckConfigMetadata(metadata *etcdraft.ConfigMetadata) error {
 	return nil
 }
 
-func validateCert(pemData []byte, certRole string) error {
-	bl, _ := pem.Decode(pemData)
+func parseCertificateFromBytes(cert []byte) (*x509.Certificate, error) {
+	pemBlock, _ := pem.Decode(cert)
+	if pemBlock == nil {
+		return &x509.Certificate{}, errors.Errorf("no PEM data found in cert[% x]", cert)
+	}
 
-	if bl == nil {
-		return errors.Errorf("%s TLS certificate is not PEM encoded: %s", certRole, string(pemData))
+	certificate, err := x509.ParseCertificate(pemBlock.Bytes)
+	if err != nil {
+		return nil, errors.Errorf("%s TLS certificate has invalid ASN1 structure %s", err, string(pemBlock.Bytes))
+	}
+
+	return certificate, nil
+}
+
+func parseCertificateListFromBytes(certs [][]byte) ([]*x509.Certificate, error) {
+	var certificateList []*x509.Certificate
+
+	for _, cert := range certs {
+		certificate, err := parseCertificateFromBytes(cert)
+		if err != nil {
+			return certificateList, err
+		}
+
+		certificateList = append(certificateList, certificate)
+	}
+
+	return certificateList, nil
+}
+
+func getCertFingerprint(cert *x509.Certificate) (string, error) {
```

Comment:
Why do you need to "compress" the certificate into a short string? You use it as a key in a map that is actually used as a set. 

You can just "cast" the certificate to string instead and check if the certificate (as a string) itself exits in a map. No need to hash it. 

```
set := make(map[string]struct{})
_, exists := set[string(certificate)]
```

Reply:
np